.github/workflows/go-tests.yml

```diff
@@ -1,17 +1,47 @@
-name: Tests
+name: Build services
 
 on:
-  workflow_call:
+  push:
+    paths:
+      - "cmd/users/**"
+      - "cmd/stellar-dominion/**"
+      - "pkg/**"
+      - "internal/**"
+      - ".github/workflows/go-tests.yml"
+      - ".github/workflows/services*.yml"
+      - ".github/workflows/user-service*.yml"
+      - ".github/workflows/stellar-dominion-service*.yml"
 
 jobs:
   tests:
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
       - name: Setup Go
         uses: actions/setup-go@v5
         with:
           go-version: "1.22.x"
-      - name: Run tests
-        run: go test ./...
+      - name: Run tests with coverage
+        run: go test ./... -coverpkg=./... -race -covermode=atomic -coverprofile=coverage.out
```

Comment:
We can restore the coverage being done alongside the (now unique) execution of the tests. The `go-tests-coverage.yml` workflow can be removed again.